src/main.c

```diff
@@ -69,11 +97,21 @@ static void on_installer_changed(GDBusProxy *proxy, GVariant *changed,
 
 	g_mutex_lock(&args->status_mutex);
 	if (g_variant_lookup(changed, "Operation", "&s", &message)) {
-		g_queue_push_tail(&args->status_messages, g_strdup(message));
+		g_queue_push_tail(&args->status_messages, g_strdup_printf("%s\n", message));
 	} else if (g_variant_lookup(changed, "Progress", "(i&si)", &percentage, &message, &depth)) {
-		g_queue_push_tail(&args->status_messages, g_strdup_printf("%3"G_GINT32_FORMAT "%% %s", percentage, message));
+		g_autofree gchar *progress = get_progress(percentage);
```

Comment:
You should only generate the progress line when needed (move it into the if block).

Reply:
Fixed.